modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/TypedInput.es.js

```diff
@@ -42,13 +40,98 @@ class TypedInput extends React.Component {
 				selected={value}
 			/>;
 	}
+	_handleDateChange = dateObject => {
+		const {onChange} = this.props;
+		const newValue = dateObject.toISOString();
+
+		onChange(newValue, PROPERTY_TYPES.DATE);
+	}
+	_renderDateType = () => {
+		const {value} = this.props;
+
+		return (<div className="criterion-input">
+			<DatePicker
+				className="form-control"
+				onChange={this._handleDateChange}
+				selected={new Date(value)}
+			/>
+		</div>);
+	}
+	_renderBooleanType = () => {
+		const {value} = this.props;
+
+		return (
+			<ClaySelect
+				className="criterion-input form-control"
+				data-testid="options-boolean"
+				onChange={this._handleChange}
+				options={[
+					{
+						label: 'TRUE',
+						value: 'true'
+					},
+					{
+						label: 'FALSE',
+						value: 'false'
+					}
+				]}
+				selected={value}
+			/>);
+	}
+	_handleNumberChange = event => {
+		const {onChange} = this.props;
+		const value = parseInt(event.target.value, 10);
+
+		if (onChange && !isNaN(value)) {
+			onChange(value.toString());
+		}
+	}
+	_renderNumberType = () => {
+		const {value} = this.props;
+
+		return (
+			<input
+				className="criterion-input form-control"
+				data-testid="simple-number"
+				onChange={this._handleNumberChange}
+				type="number"
+				value={value}
+			/>);
+	}
+	render() {
+		const {type} = this.props;
+
+		let render;
+		switch (type) {
+		case PROPERTY_TYPES.STRING:
+			render = this._renderStringType();
+			break;
+		case PROPERTY_TYPES.DATE:
+			render = this._renderDateType();
+			break;
+		case PROPERTY_TYPES.BOOLEAN:
+			render = this._renderBooleanType();
+			break;
+		case PROPERTY_TYPES.NUMBER:
+			render = this._renderNumberType();
+			break;
+		default:
+			render = null;
+		}
+		return render;
```

Comment:
Insteead of a `switch` statement, what are your thoughts on instead having a map of functions to reference? Might be more maintainable.

```javascript
	_getInputRenderer = () => {
		const {type} = this.props;

		const renderers = {
			[PROPERTY_TYPES.BOOLEAN]: this._renderBooleanType,
			[PROPERTY_TYPES.DATE]: this._renderDateType,
			[PROPERTY_TYPES.NUMBER]: this._renderNumberType,
			[PROPERTY_TYPES.STRING]: this._renderStringType
		};

		return renderers[type] || this._renderStringType;
	}

	render() {
		const inputRenderer = this._getInputRenderer();

		return inputRenderer();
	}
```